[BUG] `wasmcloud-core` does not build with default feature flags

### Affected project(s)

- [ ] documentation
- [ ] examples
- [ ] wasmCloud host
- [ ] wasmCloud CLI (wash)
- [ ] wasmCloud dashboard UI (washboard)
- [ ] capability providers
- [ ] provider bindgen
- [ ] control interface client
- [X] other / not sure

### Describe the bug

```console
~/github.com/wasmcloud/wasmcloud/crates/core ➜ cargo build
   Compiling wrpc-transport-nats v0.21.0
   Compiling oci-distribution v0.9.4
error[E0433]: failed to resolve: could not find `Certificate` in `reqwest`
   --> /Users/brooks/.cargo/registry/src/index.crates.io-6f17d22bba15001f/oci-distribution-0.9.4/src/client.rs:204:54
    |
204 |                 CertificateEncoding::Der => reqwest::Certificate::from_der(c.data.as_slice())?,
    |                                                      ^^^^^^^^^^^ could not find `Certificate` in `reqwest`
    |
note: found an item that was configured out
   --> /Users/brooks/.cargo/registry/src/index.crates.io-6f17d22bba15001f/reqwest-0.11.27/src/lib.rs:345:19
    |
345 |     pub use tls::{Certificate, Identity};
    |                   ^^^^^^^^^^^
    = note: the item is gated behind the `__tls` feature

error[E0433]: failed to resolve: could not find `Certificate` in `reqwest`
   --> /Users/brooks/.cargo/registry/src/index.crates.io-6f17d22bba15001f/oci-distribution-0.9.4/src/client.rs:205:54
    |
205 |                 CertificateEncoding::Pem => reqwest::Certificate::from_pem(c.data.as_slice())?,
    |                                                      ^^^^^^^^^^^ could not find `Certificate` in `reqwest`
    |
note: found an item that was configured out
   --> /Users/brooks/.cargo/registry/src/index.crates.io-6f17d22bba15001f/reqwest-0.11.27/src/lib.rs:345:19
    |
345 |     pub use tls::{Certificate, Identity};
    |                   ^^^^^^^^^^^
    = note: the item is gated behind the `__tls` feature

error[E0599]: no method named `danger_accept_invalid_certs` found for struct `ClientBuilder` in the current scope
   --> /Users/brooks/.cargo/registry/src/index.crates.io-6f17d22bba15001f/oci-distribution-0.9.4/src/client.rs:193:14
    |
192 |           let mut client_builder = reqwest::Client::builder()
    |  __________________________________-
193 | |             .danger_accept_invalid_certs(config.accept_invalid_certificates);
    | |             -^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `ClientBuilder`
    | |_____________|
    |

Some errors have detailed explanations: E0433, E0599.
For more information about an error, try `rustc --explain E0433`.
error: could not compile `oci-distribution` (lib) due to 3 previous errors
```

### Steps to reproduce

1. Clone repository
2. `cd crates/core`
3. `cargo update`
4. `cargo build`

### Expected behavior

I would expect this to build, like it does when default features are off and only the otel feature is specified (which is what's set up in the workspace Cargo.toml)

### Environment

- OS: [e.g. Linux, MacOS] MacOS
- Shell [e.g. bash, zsh, powershell] zsh
- wasmcloud Version [e.g. 0.81.0] v1.0.2


### Screenshots / Logs / Additional context

This appears to be an upstream compilation error in `oci-distribution` which, with our combination of feature flags

--- a/crates/core/Cargo.toml
+++ b/crates/core/Cargo.toml
@@ -33,7 +33,7 @@ hyper-rustls = { workspace = true, features = [
 ], optional = true }
 hyper-util = { workspace = true, optional = true }
 nkeys = { workspace = true }
-oci-distribution = { workspace = true, optional = true }
+oci-distribution = { workspace = true, features = ["rustls-tls"], optional = true }
 once_cell = { workspace = true }
 reqwest = { workspace = true, features = ["rustls-tls"], optional = true }
 rustls = { workspace = true, features = ["std"] }

